cql3/functions/aggregate_fcts.cc

```diff
@@ -219,14 +219,15 @@ struct aggregate_type_for<simple_date_native_type> {
 
 template<>
 struct aggregate_type_for<timeuuid_native_type> {
-    using type = timeuuid_native_type::primary_type;
+    using type = timeuuid_native_type;
 };
 
 template<>
 struct aggregate_type_for<time_native_type> {
     using type = time_native_type::primary_type;
 };
 
+// WARNING: never invoke this on temporary values; it will return a dangling reference.
```

Comment:
We can enforce it: add

```c++
template <typename Type1, typename Type2>
void max_wrapper(Type1&& t1, Type&& t2) = delete;
```

can be a nice follow-up.

Reply:
Won't that prevent invoking max_wrapper on lvalues?
https://godbolt.org/z/46E678

If it were easy to prevent exactly the bad scenario, wouldn't std::max have it already?

Reply:
Actually, it may be correct as long as you don't capture the result by reference. C++ extends the lifetime of temporaries until the end of the expression, so `auto m = std::max(temp1(), temp2());` should work (but `auto& m = std::max(temp1(), temp2());` would not).

Reply:
You're right.  I was aiming to replicate [this warning](https://en.cppreference.com/w/cpp/algorithm/max#Notes) (see "Notes" section), but I didn't capture it precisely.